### ISSUE TITLE
Cannot move immediate to memory address

### ISSUE THREAD

**Issue by andybui01** [0xADE1A1DE/AssemblyLine#12]
The Intel manual has opcodes for moving immediate values into memory, however, but it seems as though AssemblyLine currently only supports registers when MOV'ing (i.e. "mov [addr], imm" is not supported).

When moving an immediate into an address pointed to by a register, however, AssemblyLine _quietly_ continues execution.
```C
int value = 0;
assemble_str(al, "mov r8, &value\n"); // Put address of "value" into r8
assemble_str(al, "mov [r8], 1\n"); // is outputted as "mov r8, 1" by AssemblyLine
.
.
.

// Run above code
func();

printf("Value is %d\n, value); // 0
```

The current work around is to mov the immediate into a register, and moving that into memory address:
```C
assemble_str(al, "mov r8, &value\n"); // Put address of "value" into r8
assemble_str(al, "mov r9, 1\n");
assemble_str(al, "mov [r8], r9\n");
```

**Comment by dderjoel** [comment]
So what exactly do we expect here?
You mention "quietly". When would you expect an error, but don't get it?

I also cannot reproduce the issue using asmline.
```
$ asmline <<< "mov r8, &value"
assembyline: illegal operand : "mov"
assembyline: syntax error
failed to assemble instruction: mov r8, &value
```
So that is what I'd expect to happen, because `&value` is not valid. How would assemblyline know, where this `value` is ?
Could you give a full MWE?

**Comment by andybui01** [comment]
Sorry, that was a bit of pseudocode that was left in the example, the actual example (and where I'd expect an error) is here:

```C
int value = 0;

char asm_str[50];
sprintf(asm_str, "mov r8, %p\n", &value);

assemble_str(al, asm_str);
assemble_str(al, "mov [r8], 1"); 
/*
Problematic line^
Expect an error here, instead AssemblyLine quietly outputs the following asm:
mov r8, 1
*/

assemble_str(al, "ret");

int (*func)() =(int (*)())(asm_get_code(al));

int result = func();

printf("Value is %d\n", value); // 0
```

**Comment by daviduwu9** [comment]
At the moment AssemblyLine will automatically assume the operand size specified by the register in the memory operand. ie. "[rax]" will automatically be interpreted as "qword [rax]" and "[eax]" as "dword [eax]". For now we did not implemented error checking for ambiguous operand size.

**Comment by dderjoel** [comment]
okay, I can now reproduce. 
so the bug is that `mov [r8], 1` gets interpreted as `mov r8, 1`. 
And we don't want an error here, right? We want the machine code for that like so, right?
```
$ ./tools/asmline <<< 'mov r8,1' -p  
49 c7 c0 01 00 00 00 
$ ./tools/asmline <<<'mov [r8],1' -p
49 c7 00 01 00 00 00 
```

If that is the issue, than it should be fixed in the upcoming version `assemblyline 1.1.0`, which we will soon release.

If that is not the issue, please explain which code you would be expecting.

**Comment by dderjoel** [comment]
@andybui01 we [just released v1.1.0](https://github.com/0xADE1A1DE/AssemblyLine/releases/tag/v1.1.0), which should include a fix for this issue. Can you try and report, if that worked?
